diff -Naur yast2-devtools-4.2.6-orig/build-tools/rpm/macros.yast yast2-devtools-4.2.6/build-tools/rpm/macros.yast
--- yast2-devtools-4.2.6-orig/build-tools/rpm/macros.yast	2020-10-16 23:25:10.668842537 +0200
+++ yast2-devtools-4.2.6/build-tools/rpm/macros.yast	2020-10-16 23:26:35.264763277 +0200
@@ -120,8 +120,8 @@
             %suse_update_desktop_file -d ycc_${d%.desktop} ${d%.desktop} \
         done \
     %else  # 0%%{?suse_version} \
-        %{_bindir}/find %{buildroot}/%{yast_desktopdir}/ -print0 \\\
-            -name '*.desktop' | %{_bindir}/sort -uz | \\\
+        %{_bindir}/find %{buildroot}/%{yast_desktopdir}/ -name '*.desktop' \\\
+            -print0 | %{_bindir}/sort -uz | \\\
             %{_bindir}/xargs -0 %{_bindir}/desktop-file-validate \
     %endif # 0%%{?suse_version}
 
