diff -up cobbler-3.1.2/config/apache/cobbler_web.conf.httpd cobbler-3.1.2/config/apache/cobbler_web.conf
--- cobbler-3.1.2/config/apache/cobbler_web.conf.httpd	2020-05-27 02:26:44.000000000 -0600
+++ cobbler-3.1.2/config/apache/cobbler_web.conf	2020-07-07 21:12:53.942577055 -0600
@@ -16,8 +16,6 @@ WSGIDaemonProcess cobbler_web display-na
 WSGIProcessGroup cobbler_web
 WSGIPassAuthorization On
 
-<VirtualHost *:443>
-
 <Directory "/usr/share/cobbler/web/">
         <IfModule mod_ssl.c>
             SSLRequireSSL
@@ -42,5 +40,3 @@ WSGIPassAuthorization On
         AllowOverride None
         Require all granted
 </Directory>
-
-</VirtualHost>
